### PR TITLE
Es7 store

### PR DESCRIPTION
**Pull request by ereteog**
> Close https://github.com/threatgrid/iroh/issues/4427

> Related #https://github.com/threatgrid/ctia/pull/1013

This PR pushes again the ES7 store that we reverted due to SocketException thrown during `lein test` https://github.com/threatgrid/iroh/issues/4427.
Indeed the problem was not directly related to the ES7 store but was emphasized by it. The ES connection managers were never shut down while being more numerous in this PR. 
There is 2 commits:
* the [first one](https://github.com/threatgrid/ctia/pull/1039/commits/ec84f1a43bab4bc892d0f6b2aae065467234bc21) with contains the changes of previous PR implementing es7 store.
* the [second one](https://github.com/threatgrid/ctia/pull/1039/commits/f17c5399d9603e306efb097005f737b6929f0894) implements the `stop` funciton in the StoreService

For those who already reviewed the previous PR, you can directly jump to the [second commit](https://github.com/threatgrid/ctia/pull/1039/commits/ec84f1a43bab4bc892d0f6b2aae065467234bc21).

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

### CODE REVIEW

File: test/ctia/test_helpers/crud.clj [hunk 1]
```diff
@@ -148,10 +148,10 @@
       (testing "testing wait_for values on entity revocation"
         (let [test-revoke (fn [wait_for msg]
                             (let [entity-id (new-entity-short-id)
-                                  es-index-uri-pattern (re-pattern (str ".*9200.*" entity-id ".*"))
+                                  es-index-uri-pattern (re-pattern (str ".*920.*" entity-id ".*"))
                                   path (cond-> (format "ctia/%s/%s/expire" entity entity-id)
                                          (boolean? wait_for) (str "?wait_for=" wait_for))]
-                              (with-global-fake-routes {es-index-uri-pattern {:put simple-handler}}
+                              (with-global-fake-routes {es-index-uri-pattern {:post simple-handler}}
```
Comment (ereteog) [inline test/ctia/test_helpers/crud.clj:154]: uses new [ductile update doc](https://github.com/threatgrid/ductile/blob/f8b580f60f7f63554b8f12669b4d91398e75fdbc/src/ductile/document.clj#L236) function.

File: src/ctia/entity/identity.clj [hunk 2]
```diff
@@ -85,6 +86,9 @@
      :groups em/token}}})
 
 (defrecord IdentityStore [state]
+  IStore
+  (close [this]
+    (close-cm! (:state this)))
```
Comment (ereteog) [inline src/ctia/entity/identity.clj:91]: I do not know why yet but IdentityStore is has the schema:
``` clojure
{:state StoreSchema}
```
while the other stores have the schema StoreSchema

File: src/ctia/store.clj [hunk 3]
```diff
@@ -19,7 +20,8 @@
 (defprotocol IIdentityStore
   (read-identity [this login])
   (create-identity [this new-identity])
-  (delete-identity [this org-id role]))
+  (delete-identity [this org-id role])
+  (close-identity [this]))
```
Comment (frenchy64) [inline src/ctia/store.clj:24]: `close-identity` is unused?

**Review by frenchy64 — APPROVED** [review]
Very nice, thanks for the detective work!

**Review by rplevy — APPROVED** [review]

File: src/ctia/store.clj (hunk 3 above)
Comment (ereteog) [reply src/ctia/store.clj:24]: yes, remaining trace of thought :-)

**Pull request merged**